docs/src/Parser/FileReflector.php

```diff
@@ -0,0 +1,264 @@
+				break;
+		}
+
+		// Pick up DocBlock from non-documentable elements so that it can be assigned
+		// to the next hook if necessary. We don't do this for name nodes, since even
+		// though they aren't documentable, they still carry the docblock from their
+		// corresponding class/constant/function/etc. that they are the name of. If
+		// we don't ignore them, we'll end up picking up docblocks that are already
+		// associated with a named element, and so aren't really from a non-
+		// documentable element after all.
+		if (
+			! $this->isNodeDocumentable( $node )
+			&& 'Name' !== $node->getType()
+			&& ( null !== $node->getDocComment() )
+		) {
+			$this->last_doc = $node->getDocComment();
+		}
+	}
+
+	/**
+	 * Assign queued hooks to functions and update the node stack on leaving a
+	 * node.
+	 *
+	 * We can now access the function/method reflectors, so we can assign any
+	 * queued hooks to them. The reflector for a node isn't created until the
+	 * node is left.
+	 *
+	 * @param Node $node Node that is being left.
+	 */
+	public function leaveNode( Node $node ) {
+		parent::leaveNode( $node );
+
+		switch ( $node->getType() ) {
+			case 'Stmt_Class':
+				$class = end( $this->classes );
+				if ( ! empty( $this->method_uses_queue ) ) {
+					/** @var Reflection\ClassReflector\MethodReflector $method */
+					foreach ( $class->getMethods() as $method ) {
+						if ( isset( $this->method_uses_queue[ $method->getName() ] ) ) {
+							if ( isset( $this->method_uses_queue[ $method->getName() ]['methods'] ) ) {
+								/*
+								 * For methods used in a class, set the class on the method call.
+								 * That allows us to later get the correct class name for $this, self, parent.
+								 */
+								foreach ( $this->method_uses_queue[ $method->getName() ]['methods'] as $method_call ) {
+									/** @var MethodCallReflector $method_call */
+									$method_call->set_class( $class );
+								}
+							}
+
+							$method->uses = $this->method_uses_queue[ $method->getName() ];
+						}
+					}
+				}
+
+				$this->method_uses_queue = [];
+				array_pop( $this->location );
+				break;
+
+			case 'Stmt_Function':
+				$location = array_pop( $this->location );
+				if ( property_exists( $location, 'uses' ) ) {
+					end( $this->functions )->uses = $location->uses;
+				}
+				break;
+
+			case 'Stmt_ClassMethod':
+				$method = array_pop( $this->location );
+
+				/*
+				 * Store the list of elements used by this method in the queue. We'll
+				 * assign them to the method upon leaving the class (see above).
+				 */
+				if ( ! empty( $method->uses ) ) {
+					$this->method_uses_queue[ $method->name ] = $method->uses;
+				}
+				break;
+		}
+	}
+
+	/**
+	 * Check whether a given node is a filter.
+	 *
+	 * @param Node $node Node to check.
+	 *
+	 * @return bool Whether the node is a filter.
+	 */
+	protected function isFilter( Node $node ) {
+		// Ignore variable functions.
+		if ( 'Name' !== $node->name->getType() ) {
```

Comment:
Should this not be `$node->getType()`?

![image](https://user-images.githubusercontent.com/134745/91347026-89216800-e796-11ea-95b3-6ec795313619.png)